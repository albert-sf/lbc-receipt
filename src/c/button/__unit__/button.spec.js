import { createElement } from 'lwc';
import Element from 'c/button';

const createButton = (props = {}) => {
    const element = createElement('c-button', { is: Element });
    Object.assign(element, props);
    document.body.appendChild(element);
    return element;
};

describe('c-button attributes', () => {
    it('should set title attribute on the button', () => {
        const title = 'Click here';
        const element = createButton({ title });

        const btn = element.shadowRoot.querySelector('button');

        expect(btn.getAttribute('title')).toBe(title);
    });
});
