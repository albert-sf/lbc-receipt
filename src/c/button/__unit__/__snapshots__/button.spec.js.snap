// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-button should render 1`] = `
<c-button
  title="Click here"
>
  #shadow-root(open)
    <button
      title="Click here"
    />
</c-button>
`;
